pkg/apis/etcd/v1beta2/restore_types.go

```diff
@@ -43,8 +43,26 @@ type RestoreSpec struct {
 	// using this spec, restore operator will prepare the seed that
 	// etcd operator will pick up later.
 	ClusterSpec ClusterSpec `json:"clusterSpec"`
+
+	// BackupStorageType is the type of the backup storage which is used as RestoreSource.
+	// TODO: implement me.
+	BackupStorageType string `json:"backupStorageType"`
 	// RestoreSource tells the where to get the backup and restore from.
 	RestoreSource `json:",inline"`
+	// EtcdCluster references an EtcdCluster resource which should have "Failed" status.phase
+	// and to be restored.
+	// TODO: implement me.
+	EtcdCluster EtcdClusterRef `json:"etcdCluster"`
+}
+
+// EtcdClusterRef tells the EtcdCluster which should have "Failed" status.phase
+// and to be restored.
+type EtcdClusterRef struct {
```

Comment:
who creates `EtcdCluster` cr?
previously, restore operator tooks in a cluster spec and create an EtcdCluster CR and gives that to etcd operator.
suppose that I just want to use restore operator to create a new etcd cluster with some backup.  What should I specify in this new `RestoreSpec`?

Reply:
> who creates EtcdCluster cr?

Restore operator will create it.

> What should I specify in this new RestoreSpec?

The EtcdCluster resource. For example, an EtcdCluster should have failed (although not necessary, but fit into restore assumption), then we just reference it in Restore spec.

Reply:
so we specify the name of EtcdCluster in RestoreSpec.
then restore operator creates the EtcdCluster CR based on `EtcdClusterRef.Name`.

Reply:
Correct.
Note that it also has a `Namespace`. So we can also migrate EtcdCluster from another namespace too. In the same namespace, we expect only "Failed" ones.

Reply:
`In the same namespace, we expect only "Failed" ones.`
this might not always be true right? in the simple case, i just want to create a new cluster with a snapshot. there is no failed cluster anywhere.

If the is a failed `clusterA`, I want to restore `clusterA` with the same name using etcd restore operator. is seems to me that's where ` EtcdCluster which should have "Failed" status.phase`. How do we deal that then?

Reply:
Let's assume this is true for now. We can revisit this after Feb.

> in the simple case, i just want to create a new cluster with a snapshot.
That sounds interesting use case. Can you add that as a feature request issue?

> How do we deal that then?
That's implementation detail. We can discuss it later.
Haseeb will take care of it. Let's not block him.